developer-docs-site/docs/guides/sign-a-transaction.md

```diff
@@ -0,0 +1,311 @@
+:::
+
+## Overview
+
+Creating signed transactions can be splited into four steps: **preparing the raw transaction (unsigned)**, **getting signing message by serializing the raw transaction**, **signing the serialized transaction to get a signature**, and **deriving a signed transaction from raw transaction and signature**.
+
+Raw transactions are the transactions that have not been signed yet. Raw transactions contain the information about what needs to be executed (transaction payload) and the maximum gas this transaction can consume. A signing message is contructed with a prefix and a body. The body is the serialized bytes of raw transactions by applying the Binary Canonical Serialization ([BCS][bcs]). The signing step is a standard [Ed25519][ed25519] encryption of the signing message with the private key provided by clients. Signing generates a signature that represents the authorization by the client for the chain to run the transaction. Finally, clients need to compose a signed transaction with the original raw transaction and the signature.
+## Key Concepts
+
+### Raw transaction
+Raw transactions we described in this guide means the data input we would submit to the Aptos blockchain for execution. In details, raw transactions are consisted of following data fields:
+* **sender** hex-coded account address of the sender of the transaction
+* **sequence_number** Sequence number of this transaction. This must match the sequence number stored in the sender's account at the time the transaction executes.
+*  **payload** The actual work that the Aptos blockchain needs to do.
+*  **max_gas_amount** Maximal total gas to spend for this transaction.
+*  **gas_unit_price** Price to be paid per gas unit.
+*  **expiration_timestamp_secs** If the current blockchain timestamp is greater than or equal to this time, then the transaction has expired and will be discarded.
+*  **chain_id** The chain id that this transaction is targeting to be ran on.
+
+### BCS
+Binary Canonical Serialization (BCS) is a serialization format. The design goals of BCS include:
+* provide good performance and concise (binary) representations
+* enforce canonical serialization, meaning that every value of a given type should have a single valid representation.
+* support a wide range of data types, including booleans, integers, strings, lists, maps and user-defined structs, etc. Note: floats, unicode chars and sets are not supported.
+
+BCS is not a self-describing format. As such, in order to deserialize a message, one must know the message type and layout ahead of time.
+
+An example of how BCS serializes a string.
+```typescript
+// A string is serialized as: byte length + byte representation of the string. The byte length is required for deserialization. Without it, no way the deserializer knows how many bytes to deserialize.
+const bytes: Unint8Array = bcs_serialize_string("aptos");
+assert(bytes == [0x61, 0x70, 0x74, 0x6F, 0x73]);
```

Comment:
this isn't right, it should be 0x05 first followed by the 5 bytes.

Reply:
👍

Reply:
😂